ballerina/client.bal

```diff
@@ -256,6 +256,70 @@ public isolated client class Client {
         http:Response httpResponse = check self.amazonS3->delete(requestURI, request);
         return handleHttpResponse(httpResponse);
     }
+    
+    # Generates a presigned URL for the object.
+    #
+    # + bucketName - The name of the bucket  
+    # + objectName - The name of the object  
+    # + httpMethod - The HTTP method to be used, either GET or PUT  
+    # + expires - The time period for which the presigned URL is valid, in seconds  
+    # + partNo - The part number of the object, when uploading multipart objects  
+    # + uploadId - The upload ID of the multipart upload  
+    # + objectCreationHeaders - Optional headers for GET http method 
+    # + objectRetrievalHeaders - Optional headers for PUT http method 
+    # + return - If successful, a presigned URL, else an error
+    @display {label: "Create Presigned URL"}
+    remote isolated function createPresignedUrl(
+            @display {label: "Bucket Name"} string bucketName,
+            @display {label: "Object Name"} string objectName,
+            @display {label: "HTTP Method"} GET|PUT httpMethod,
+            @display {label: "Expiration Time"} int expires = 1800,
+            @display {label: "Object Creation Headers"} ObjectCreationHeaders? objectCreationHeaders = (),
+            @display {label : "Object Retrieval Headers"} ObjectRetrievalHeaders? objectRetrievalHeaders = (),
```

Comment:
These are tightly bound to the method parameter, right? And will always be one of the two? 

I feel like we need to rethink the types here. E.g., wouldn't something like

```ballerina
GET|PUT|record {| GET method = GET; ObjectRetrievalHeaders objectRetrievalHeaders; |}|record {| PUT method = PUT; ObjectCreationHeaders objectCreationHeaders; |} action;
```

make more sense here?


Can call with

```ballerina
createPresignedUrl(bn, obn, s3:GET, ...);
createPresignedUrl(bn, obn, s3:PUT, ...);
createPresignedUrl(bn, obn, {objectRetrievalHeaders: {...}}, ...); // method is automatically GET
createPresignedUrl(bn, obn, {objectCreationHeaders: {...}}, ...); // method is automatically PUT
```

@NipunaRanasinghe do we have a standard we follow for such cases?

Reply:
Would it then be ideal to define two records for the two methods?

```ballerina
public type putObjectWithHeaders record {
    PUT method = PUT;
    ObjectCreationHeaders headers;
};

public type getObjectWithHeaders record {
    GET method = GET;
    ObjectRetrievalHeaders headers;
};
```

and then extract the method like this 

```ballerina
GET|PUT httpMethod;

if action is GET|PUT {
    httpMethod = action;
} else if action is getObjectWithHeaders|putObjectWithHeaders {
    httpMethod = action.method;
}
```

and go ahead as usual?

Reply:
Apologies for the late response. I also believe that the suggested approach looks better, given that the headers are tightly coupled with the action in this case. 

@nisuraaa I'm also +1 to have named records (instead of anonymous records) to represent the combination of action and headers, but I'm concerned with the proposed names. 

```ballerina
public type PUTWithHeaders record {
    PUT method = PUT;
    ObjectCreationHeaders headers;
};

public type GETWithHeaders record {
    GET method = GET;
    ObjectRetrievalHeaders headers;
};
```